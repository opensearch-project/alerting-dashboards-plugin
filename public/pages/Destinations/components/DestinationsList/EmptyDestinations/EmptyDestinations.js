/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import PropTypes from 'prop-types';
import { EuiButton, EuiEmptyPrompt, EuiLink, EuiText } from '@elastic/eui';
import { MANAGE_CHANNELS_PATH } from '../../../../CreateTrigger/utils/constants';

const filterText = (hasNotificationPlugin) =>
  hasNotificationPlugin ? (
    <>
      <p>
        There are no destinations matching your applied filters. Reset your filters to view all
        destinations.
      </p>
      <p>
        Migrated destinations can be found in&nbsp;
        {<EuiLink href={MANAGE_CHANNELS_PATH}>Notifications</EuiLink>}
      </p>
    </>
  ) : (
    <p>
      There are no destinations matching your applied filters. Reset your filters to view all
      destinations.
    </p>
  );

const emptyText = <p>There are no existing destinations.</p>;

const resetFiltersButton = (resetFilters) => (
  <EuiButton fill onClick={resetFilters}>
    Reset filters
  </EuiButton>
);

const propTypes = {
  isFilterApplied: PropTypes.bool.isRequired,
  onResetFilters: PropTypes.func.isRequired,
};

const EmptyDestinations = ({ hasNotificationPlugin, isFilterApplied, onResetFilters }) => (
  <EuiEmptyPrompt
    style={{ maxWidth: '45em' }}
    body={<EuiText>{isFilterApplied ? filterText(hasNotificationPlugin) : emptyText}</EuiText>}
    actions={isFilterApplied ? resetFiltersButton(onResetFilters) : undefined}
  />
);

EmptyDestinations.propTypes = propTypes;

export default EmptyDestinations;
