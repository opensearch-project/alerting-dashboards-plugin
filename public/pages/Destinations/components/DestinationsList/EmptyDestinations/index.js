/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import EmptyDestinations from './EmptyDestinations';

export default EmptyDestinations;
