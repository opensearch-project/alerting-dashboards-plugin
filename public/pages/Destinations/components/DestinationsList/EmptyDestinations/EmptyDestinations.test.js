/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { mount, render } from 'enzyme';
import EmptyDestinations from './EmptyDestinations';

describe('<EmptyDestinations />', () => {
  test('should render empty destinations message', () => {
    const wrapper = render(
      <EmptyDestinations isFilterApplied={false} onResetFilters={jest.fn()} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('should render no results for filter criteria', () => {
    const wrapper = render(<EmptyDestinations isFilterApplied onResetFilters={jest.fn()} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should call reset Filters callback on click of Reset Filters Button', () => {
    const handleResetFilter = jest.fn();
    const wrapper = mount(<EmptyDestinations isFilterApplied onResetFilters={handleResetFilter} />);
    // Simulate Reset button click Click
    wrapper.find('button').simulate('click');
    expect(handleResetFilter).toHaveBeenCalledTimes(1);
  });
});
