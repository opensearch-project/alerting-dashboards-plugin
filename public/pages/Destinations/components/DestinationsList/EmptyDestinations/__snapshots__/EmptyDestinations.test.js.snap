// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmptyDestinations /> should render empty destinations message 1`] = `
<div
  class="euiEmptyPrompt"
  style="max-width:45em"
>
  <span
    class="euiTextColor euiTextColor--subdued"
  >
    <div
      class="euiText euiText--medium"
    >
      <div
        class="euiText euiText--medium"
      >
        <p>
          There are no existing destinations.
        </p>
      </div>
    </div>
  </span>
</div>
`;

exports[`<EmptyDestinations /> should render no results for filter criteria 1`] = `
<div
  class="euiEmptyPrompt"
  style="max-width:45em"
>
  <span
    class="euiTextColor euiTextColor--subdued"
  >
    <div
      class="euiText euiText--medium"
    >
      <div
        class="euiText euiText--medium"
      >
        <p>
          There are no destinations matching your applied filters. Reset your filters to view all destinations.
        </p>
      </div>
    </div>
  </span>
  <div
    class="euiSpacer euiSpacer--l"
  />
  <button
    class="euiButton euiButton--primary euiButton--small euiButton--fill"
    type="button"
  >
    <span
      class="euiButtonContent euiButton__content"
    >
      <span
        class="euiButton__text"
      >
        Reset filters
      </span>
    </span>
  </button>
</div>
`;
